### ISSUE TITLE
Custom cluster creation stuck in updating state with RHEL 7.6 nodes and native docker 

### ISSUE THREAD

**Issue by soumyalj** [rancher/rancher#19305]
Rancher version : v2.1.8

K8s version: v1.13.5
Docker version : native docker 1.13.1
OS : RHEL 7.6

Steps

1. Create a 3 node custom cluster with RHEL 7.6 nodes and native docker 1.13.1, selinux enabled. Choose k8s version v1.13.5.

N1 : All roles
N2: All roles
N3: All roles

Note: The nodes were added one after another (around 10s apart)

Cluster is stuck in "Updating State"

One node gets registered and other two nodes are stuck in "registering" state. 
It was observed in the logs that initially cluster was provisioned with only one node and a cluster update was triggered with the remaining nodes. But the cluster provisioning failed 

Rancher server logs
```

2019/04/01 20:37:09 [INFO] cluster [c-z6lnt] provisioning: [worker] Successfully started [rke-log-linker] container on host [abcd]
2019/04/01 20:37:09 [INFO] cluster [c-z6lnt] provisioning: [remove/rke-log-linker] Successfully removed container on host [efgh]
2019/04/01 20:37:09 [INFO] cluster [c-z6lnt] provisioning: [remove/rke-log-linker] Successfully removed container on host [klmn]
2019/04/01 20:37:09 [INFO] cluster [c-z6lnt] provisioning: [worker] Successfully started Worker Plane..
2019/04/01 20:37:09 [INFO] cluster [c-z6lnt] provisioning: [sync] Syncing nodes Labels and Taints
2019/04/01 20:37:40 [ERROR] cluster [c-z6lnt] provisioning: Failed to get kubernetes node [ip-172-31-3-149]:  "ip-172-31-3-149" not found
2019/04/01 20:37:40 [ERROR] ClusterController c-z6lnt [cluster-provisioner-controller] failed with : Failed to get kubernetes node [ip-172-31-3-149]:  "ip-172-31-3-149" not found
2019-04-01 20:38:37.410141 I | mvcc: store.index: compact 9734
2019-04-01 20:38:37.412821 I | mvcc: finished scheduled compaction at 9734 (took 1.090702ms)

```

In an other attempt of the same use case, two nodes got registered and the third node was stuck in registering state

```
This cluster is currently Updating; areas that interact directly with it will not be available until the API is ready.

[controlPlane] Failed to bring up Control Plane: Failed to verify healthcheck: Failed to check https://localhost:6443/healthz for service [kube-apiserver] on host [ijkl]: Get https://localhost:6443/healthz: read tcp [::1]:56624->[::1]:6443: read: connection reset by peer, log: ; [healthcheck] Start Healthcheck on service [kube-apiserver] on host [efgh]


```

Rancher server logs 

```
2019/04/01 23:04:58 [INFO] cluster [c-blhps] provisioning: [network] Running control plane -> etcd port checks
2019/04/01 23:05:00 [INFO] cluster [c-blhps] provisioning: [network] Successfully started [rke-port-checker] container on host [abcd]
2019/04/01 23:05:00 [INFO] cluster [c-blhps] provisioning: [network] Successfully started [rke-port-checker] container on host [efgh]
2019/04/01 23:05:00 [INFO] cluster [c-blhps] provisioning: [network] Successfully started [rke-port-checker] container on host [ijkl]
2019/04/01 23:05:05 [ERROR] cluster [c-blhps] provisioning: [network] Host [efgh] is not able to connect to the following ports: [ijkl:2379]. Please check network policies and firewall rules
2019/04/01 23:05:05 [ERROR] ClusterController c-blhps [cluster-provisioner-controller] failed with : [network] Host [efgh] is not able to connect to the following ports: [ijkl:2379]. Please check network policies and firewall rules


```
Note: A custom cluster using ubuntu and RHEL 7.5 nodes comes up correctly even when the nodes are added almost at the same time with few seconds apart.

**Comment by soumyalj** [comment]
Custom cluster(3 node) did not come with RHEL nodes using Native docker (SELinux disabled)
Cluster was stuck in updating state. Only one node was registered and the other two nodes did not get registered.
```

This cluster is currently Updating; areas that interact directly with it will not be available until the API is ready.

Failed to get kubernetes node [ip-172-31-7-abc]: "ip-172-31-7-abc" not found
```
 Logs below:
```
2019/04/02 04:12:28 [INFO] cluster [c-9pjd6] provisioning: [sync] Syncing nodes Labels and Taints
2019/04/02 04:12:36 [ERROR] cluster [c-z6lnt] provisioning: Failed to get kubernetes node [ip-172-31-3-149]:  "ip-172-31-3-149" not found
2019/04/02 04:12:36 [ERROR] ClusterController c-z6lnt [cluster-provisioner-controller] failed with : Failed to get kubernetes node [ip-172-31-3-abc]:  "ip-172-31-3-abc" not found
2019/04/02 04:12:58 [ERROR] cluster [c-9pjd6] provisioning: Failed to get kubernetes node [ip-172-31-7-182]:  "ip-172-31-7-182" not found
2019/04/02 04:12:59 [ERROR] ClusterController c-9pjd6 [cluster-provisioner-controller] failed with : Failed to get kubernetes node [ip-172-31-7-182]:  "ip-172-31-7-182" not found
2019-04-02 04:13:37.958372 I | mvcc: store.index: compact 50472
2019-04-02 04:13:37.962001 I | mvcc: finished scheduled compaction at 50472 (took 1.622353ms)
2019/04/02 04:15:48 [ERROR] cluster [c-blhps] provisioning: [controlPlane] Failed to bring up Control Plane: Failed to verify healthcheck: Failed to check https://localhost:6443/healthz for service [kube-apiserver] on host [abcd]: Get https://localhost:6443/healthz: read tcp [::1]:53306->[::1]:6443: read: connection reset by peer, log: 
2019/04/02 04:15:48 [ERROR] ClusterController c-blhps [cluster-provisioner-controller] failed with : [controlPlane] Failed to bring up Control Plane: Failed to verify healthcheck: Failed to check https://localhost:6443/healthz for service [kube-apiserver] on host [abcd]: Get https://localhost:6443/healthz: read tcp [::1]:53306->[::1]:6443: read: connection reset by peer, log: 
2019-04-02 04:18:37.964267 I | mvcc: store.index: compact 51144
2019-04-02 04:18:37.968268 I | mvcc: finished scheduled compaction at 51144 (took 2.027574ms)



```

**Comment by soumyalj** [comment]
Follow up Observation for the above issue:

When RHEL nodes are added one at a time by allowing each node to come to active state(before adding the next one), the custom cluster gets provisioned correctly.

**Comment by alena1108** [comment]
This issue has the same root cause as the issues documented in RN for 2.1.x release line:

https://github.com/rancher/rancher/issues/15970
https://github.com/rancher/rancher/issues/15969
https://github.com/rancher/rancher/issues/15695

When custom nodes get registered in parallel, there are chances of cluster being stuck in updating/provisioning state. It is not RHEL 7.6 specific, and happens for RHEL 7.5 as well. The issue is fixed in 2.2.0 by refactoring etcd and state management for RKE clusters.
